controller/workitem_blackbox_test.go

```diff
@@ -101,6 +108,129 @@ func TestGetWorkItemWithLegacyDescription(t *testing.T) {
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result1.Data.ID)
+	_, result2 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result2.Data.ID)
+	_, result3 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result3.Data.ID)
+	_, result4 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result4.Data.ID)
+	_, result5 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result5.Data.ID)
+	payload2 := minimumRequiredReorderPayload()
+
+	// This case reorders two workitems -> result2 and result3 and places them above result1
+	// This also tests reordering of multiple work items
+
+	var dataArray []*app.WorkItem2 // dataArray contains the workitems that have to be reordered
+	dataArray = append(dataArray, result2.Data, result3.Data)
+	payload2.Data = dataArray
+	payload2.Position.ID = *result1.Data.ID // Position.ID specifies the workitem ID above or below which the workitem(s) have to be places
+	payload2.Position.Direction = above
+
+	_, reordered1 := test.ReorderWorkitemOK(t, nil, nil, controller, &payload2) // Returns the workitems which have reordered
+
+	require.Len(t, reordered1.Data, 2) // checks the correct number of workitems reordered
+
+	assert.Equal(t, result2.Data.Attributes["version"].(int)+1, reordered1.Data[0].Attributes["version"])
+	assert.Equal(t, result3.Data.Attributes["version"].(int)+1, reordered1.Data[1].Attributes["version"])
+
+	assert.Equal(t, *result2.Data.ID, *reordered1.Data[0].ID)
+	assert.Equal(t, *result3.Data.ID, *reordered1.Data[1].ID)
+
+	// Return error if order of reordered workitem is greater than the workitem position.id
+	if (reordered1.Data[0].Attributes["order"].(float64) > result1.Data.Attributes["order"].(float64)) || reordered1.Data[1].Attributes["order"].(float64) > result1.Data.Attributes["order"].(float64) {
+		t.Error("Reorder incorrect1111")
+	}
+
+	// This case reorders one workitem -> result4 and placed it below result5
+
+	// clear the dataArray
+	dataArray = dataArray[:0]
+	dataArray = append(dataArray, result4.Data)
+	payload3 := minimumRequiredReorderPayload()
+	payload3.Data = dataArray
+	payload3.Position.ID = *result5.Data.ID
+	payload3.Position.Direction = below
+	_, reordered2 := test.ReorderWorkitemOK(t, nil, nil, controller, &payload3)
+	require.Len(t, reordered2.Data, 1)
+	assert.Equal(t, result4.Data.Attributes["version"].(int)+1, reordered2.Data[0].Attributes["version"])
+	assert.Equal(t, *result4.Data.ID, *reordered2.Data[0].ID)
+
+	// Return error if order of reordered workitem is less than the workitem position.id
+	if reordered2.Data[0].Attributes["order"].(float64) < result4.Data.Attributes["order"].(float64) {
```

Comment:
What about splitting this into different asserts for easier readability? See https://github.com/almighty/almighty-core/pull/804#discussion_r103488430